src/streamlink/logger.py

```diff
@@ -17,17 +16,7 @@
 _config_lock = Lock()
 
 
-class _CompatLogRecord(logging.LogRecord):
-    """
-    LogRecord wrapper to include sinfo for Python 3 by not Python 2
-    """
-
-    def __init__(self, name, level, pathname, lineno, msg, args, exc_info, func=None, sinfo=None, **kwargs):
-        super(_CompatLogRecord, self).__init__(name, level, pathname, lineno, msg, args, exc_info, func=func,
-                                               sinfo=sinfo, **kwargs)
-
-
-class _LogRecord(_CompatLogRecord):
+class LogRecord(logging.LogRecord):
```

Comment:
This custom `LogRecord` class overrides the percent-format-style of the default implementation with `string.format()`. Until all log calls with custom LogRecord.args attributes (`log.info("foo {0}", bar)`) have been replaced with properly formatted input messages (`log.info(f"foo {bar}")`), this can't be removed, as the percent format style would be needed otherwise (`log.info("foo %s", bar)`) by those log calls.

I think the removal of `if name.startswith("streamlink"):` in `makeRecord` is therefore incorrect, as external modules using Streamlink's root logger could potentially use the default format, and this would break it.

As said, I have another branch with the replacement of all `self.logger` calls in the plugins and this also removes the additional log call attributes with properly formatted messages. There may be lots of other log calls in other Streamlink modules, and until these haven't been replaced, the custom LogRecord class can't be removed.